src/judo.js

```diff
@@ -106,10 +106,19 @@ const run = async () => {
  */
 const runStepFile = async (yamlFilePath, options) => {
   const fileContents = fs.readFileSync(yamlFilePath);
-  const yamlFile = yaml.safeLoad(fileContents, 'utf8');
-  // logger.info('Running step file: ' + yamlFilePath);
-  const runSteps = yamlFile.run;
-  const runStepNames = Object.keys(runSteps);
+  let runSteps = {};
+  // first step for every file is gonna be parse itself
+  let runStepNames = ['parseYamlFile'];
+  let parseErrorMessage;
+  try {
+    const yamlFile = yaml.safeLoad(fileContents, 'utf8');
+    // logger.info('Running step file: ' + yamlFilePath);
+    runSteps = yamlFile.run;
+    runStepNames = Object.keys(runSteps);
+  } catch (e) {
+    parseErrorMessage = e.message;
```

Comment:
If there is an error loading the file is there any benefit in continuing? Wouldn't the first execute run test throw an error? which i see you catch below, but it might make more sense to just bail here.

Reply:
I was mostly thinking about the user experience, I wanted to still show the summary for the tests, wanted to give a `stepName` to this step of checking yml file as well instead of just bail over there.

If I just bail there, I would replace `parseErrorMessage = e.message;` by 

```
logger.error(e.message);
process.exit(1);
```

the output for first approach is:

```
[INFO]     Executing run step: parseYamlFile
[ERROR]    FAILED "parseYamlFile": bad indentation of a mapping entry at line 5, column 3:
      - /On branch ZZZ-[0-9]{0,10}\/no-id/
      ^ (1ms)

[SUMMARY]  ============= JUDO TESTS COMPLETE =============
[SUMMARY]  test-examples/wrong-yml.yml
[ERROR]    FAILED: parseYamlFile (0.001s)
[SUMMARY]  Total tests: 1 (0.001s)
[SUMMARY]  Passed:      0
[SUMMARY]  Failed:      1
```

and for the second approach is:

```
[ERROR]    bad indentation of a mapping entry at line 5, column 3:
      - /On branch ZZZ-[0-9]{0,10}\/no-id/
      ^
```


@worldjoe do you think I should this second approach then? Thinking about the user, I would prefer the second but it is up to you.